zcash_client_backend/src/data_api/error.rs

```diff
@@ -79,11 +86,9 @@ pub enum Error<DataSourceError, CommitmentTreeError, SelectionError, FeeError, C
     /// Attempted to create a spend to an unsupported Unified Address receiver
     NoSupportedReceivers(Box<UnifiedAddress>),
 
-    /// A proposed transaction cannot be built because it requires spending an input
-    /// for which no spending key is available.
-    ///
-    /// The argument is the address of the note or UTXO being spent.
-    NoSpendingKey(String),
+    /// A proposed transaction cannot be built because it requires spending an input of a type for
+    /// which a key that is required to construct the transaciton is not available.
```

Comment:
```suggestion
    /// A proposed transaction cannot be built because it requires spending an input
    /// of a type for which a key required to construct the transaction is not available.
```

Reply:
This error is being generated based on what kinds of keys are present in a USK; thus the "type" of the input here is really the "pool" of the input. In particular, no consideration is being given here to transparent multisig (a use case we do not currently support) where we might have some but not all of the necessary keys, because we'd never have all of the keys in the same USK (that would defeat the point).